src/NuGet.Clients/NuGet.CommandLine/Commands/ProjectFactory.cs

```diff
@@ -236,6 +237,19 @@ public PackageBuilder CreateBuilder(string basePath, NuGetVersion version, strin
                         Path.GetFullPath(Path.GetDirectoryName(TargetPath))), LogLevel.Minimal));
             }
 
+            string usingNETSDK = _project.GetPropertyValue("UsingMicrosoftNETSDK");
+            if (!string.IsNullOrEmpty(usingNETSDK)) // NuGet.exe cannot correctly pack SDK based projects.
```

Comment:
Isn't it better to check for any imported SDKs than checking some property which can be set anywhere in the build environment?

Before you might say, customized projects are not supported, let me reiterate here, that's not a reason to ignore the correctness of the check. If we're checking for only .NET SDK, then this check is fine. But if we're checking for general SDK-style projects, then we should check whether any SDKs have been imported.

Is there an MSBuild API that offers a list of imported SDKs for a given project or a list of import elements (_so that we could essentially check for an SDK attribute or an element_)?

Reply:
> Isn't it better to check for any imported SDKs than checking some property which can be set anywhere in the build environment?

There's no easy way of doing that from an evaluation today. You can check imports, but correlating that to SDKs can be error prone. 

> If we're checking for only .NET SDK, then this check is fine. But if we're checking for general SDK-style projects, then we should check whether any SDKs have been imported

We're checking for the .NET SDK in particular, as the pack functionality is SDK specific, it's bundled with the .NET SDK projects. We can check for the pack SDK being imported, but I want to go to target a very specific set of customers.

> Isn't there an MSBuild API that offers a list of imported SDKs for a given project or a list of import elements (so that we could essentially check for an SDK attribute or an element)?

Unfortunately not possible today. You can get the imports and infer the usage from there, but this seemed simpler and helps address the target scenario.

Reply:
Why can't we just invoke `msbuild -t:Pack` here instead of displaying error? Since NuGet pack command already uses MSBuild (_for building the project_), this shouldn't be an issue, no?

Just curious.

Reply:
Addressed in the doc. 
At this point, we're moving forward with erroring.